src/features/os-config/index.ts

```diff
@@ -1,27 +1,46 @@
-import type { Application, RequestHandler } from 'express';
+import type { Application } from 'express';
 
 import {
 	DEVICE_CONFIG_OPENVPN_CA,
 	DEVICE_CONFIG_OPENVPN_CONFIG,
 	DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
+	LOGS_HOST,
 } from '../../lib/config';
 import { b64decode } from '../../lib/utils';
 
-const getOsConfiguration: RequestHandler = (_req, res) => {
-	res.json({
-		services: {
-			openvpn: {
-				config: DEVICE_CONFIG_OPENVPN_CONFIG,
-				ca: b64decode(DEVICE_CONFIG_OPENVPN_CA),
-			},
-			ssh: {
-				authorized_keys: DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
-			},
-		},
-		schema_version: '1.0.0',
-	});
+// OS service configurations
+const services = {
+	openvpn: {
+		config: DEVICE_CONFIG_OPENVPN_CONFIG,
+		ca: b64decode(DEVICE_CONFIG_OPENVPN_CA),
+	},
+	ssh: {
+		authorized_keys: DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
+	},
+};
+
+// Config.json migrations: Changes should be evaluated for security risks before applying.
+//
+// - A field may not be deleted from config.json.
+// - A field with a value of non-null tells os-config that the value will be updated
+//   to the new value if it's different.
+// - A field not found in the whitelist of the os-config schema will be ignored.
+const config = {
+	overrides: {
+		logsEndpoint: LOGS_HOST != null ? `https://${LOGS_HOST}` : undefined,
+	},
 };
 
 export const setup = (app: Application) => {
-	app.get('/os/v1/config/', getOsConfiguration);
+	app.get('/os/v1/config/', (_req, res) => {
+		res.json({
+			services,
+			// Older os-configs don't know how to handle the config field, but
+			// luckily serde-rs ignores unknown fields by default.
+			config,
+			// schema_version is an outdated field that is kept to be compatible
```

Comment:
@cywang117 perhaps we could use `/** @deprecated <comment> */` but no big deal :)